### ISSUE TITLE
How to account for singular/plural formatting?

### ISSUE THREAD

**Issue by nathanchase** [jsmreese/moment-duration-format#48]
Let's say I have the following:
`moment.duration(121, "minutes").format("h [hrs], m [min]");`
which outputs: 2 hrs, 1 min

but If I have
`moment.duration(90, "minutes").format("h [hrs], m [min]");`
then I get: 1 hrs, 30 min

Ideally, I'd get: 1 hr, 30 mins instead of 1 hrs, 30 min
How can I account for singular/plural forms of "hr/hrs" and "min/mins"?

Is there a way to regex or otherwise format to accomplish this?

**Comment by nathanchase** [comment]
This was my workaround, but I would have expected there to be a way to do this without having to do the formatting logic myself:

`convertMinutes(mins) {
        let hours = Math.floor(mins / 60);
        let minutes = mins % 60;

        if (hours < 1) {
            hours = "";
        } else if (hours == 1) {
            hours = hours + " hr, ";
        } else {
            hours = hours + " hrs, ";
        }

        if (minutes < 1) {
            minutes = "";
        } else if (minutes == 1) {
            minutes = minutes + " min";
        } else {
            minutes = minutes + " mins";
        }

        let time = hours + minutes;

        return time;
    }`

**Comment by Leeds-eBooks** [comment]
Looks like #38 will handle this.

**Comment by kenjiru** [comment]
Can you close this, so everybody can focus on #38?

**Comment by jsmreese** [comment]
I've pushed a new version of the plugin to the master branch. It's labeled as version 2.0.0b1 in the js file.
This version implements `largest` as well as `trim: "both"` (which can be used in conjunction with largest).
This version also extends the moment `en` locale so that duration unit strings will be auto-singularized. e.g. instead of outputting `1 minutes` or `1 mins`, the plugin will now automatically output `1 minute` or `1 min`.
Basic documentation for the new options can be found in the js file in the default options section. I'll add documentation to the readme in the next few days.
Once I've done that, added a few more tests, and reviewed the other open issues, I'll publish version 2.0.0.

Closing this as suggested.
